stubs/WebOb/@tests/stubtest_allowlist.txt

```diff
@@ -0,0 +1,89 @@
+webob.acceptparse.AcceptCharset.charset_re
+webob.acceptparse.AcceptEncoding.accept_encoding_compiled_re
+webob.acceptparse.AcceptEncoding.codings_n_weight_compiled_re
+webob.acceptparse.AcceptEncoding.codings_n_weight_re
+webob.acceptparse.AcceptEncoding.codings_re
+webob.acceptparse.AcceptLanguage.accept_language_compiled_re
+webob.acceptparse.AcceptLanguage.lang_range_n_weight_compiled_re
+webob.acceptparse.AcceptLanguage.lang_range_n_weight_re
+webob.acceptparse.AcceptLanguage.lang_range_re
+webob.acceptparse.MIMEAccept
+webob.acceptparse.OWS_re
+webob.acceptparse._AcceptCharsetInvalidOrNoHeader.__add__
+webob.acceptparse._AcceptCharsetInvalidOrNoHeader.__radd__
+webob.acceptparse._AcceptCharsetInvalidOrNoHeader.copy
+webob.acceptparse._AcceptCharsetInvalidOrNoHeader.parsed
+webob.acceptparse._AcceptEncodingInvalidOrNoHeader.__add__
+webob.acceptparse._AcceptEncodingInvalidOrNoHeader.__radd__
+webob.acceptparse._AcceptEncodingInvalidOrNoHeader.copy
+webob.acceptparse._AcceptEncodingInvalidOrNoHeader.parsed
+webob.acceptparse._AcceptInvalidOrNoHeader.__add__
+webob.acceptparse._AcceptInvalidOrNoHeader.__radd__
+webob.acceptparse._AcceptInvalidOrNoHeader.copy
+webob.acceptparse._AcceptInvalidOrNoHeader.parsed
+webob.acceptparse._AcceptLanguageInvalidOrNoHeader.__add__
+webob.acceptparse._AcceptLanguageInvalidOrNoHeader.__radd__
+webob.acceptparse._AcceptLanguageInvalidOrNoHeader.copy
+webob.acceptparse._AcceptLanguageInvalidOrNoHeader.lookup
+webob.acceptparse._AcceptLanguageInvalidOrNoHeader.parsed
+webob.acceptparse.qvalue_re
+webob.acceptparse.tchar_re
+webob.acceptparse.token_compiled_re
+webob.acceptparse.token_re
+webob.acceptparse.weight_re
+webob.cachecontrol.CacheControl.max_age
+webob.cachecontrol.CacheControl.max_stale
+webob.cachecontrol.CacheControl.min_fresh
+webob.cachecontrol.CacheControl.s_max_age
+webob.cachecontrol.CacheControl.s_maxage
+webob.cachecontrol.CacheControl.stale_if_error
+webob.cachecontrol.CacheControl.stale_while_revalidate
+webob.cachecontrol.CacheControl.update_dict
+webob.cachecontrol.UpdateDict.setdefault
+webob.cachecontrol.need_quote_re
+webob.cachecontrol.token_re
+webob.client.SendRequest.MULTILINE_RE
+webob.compat
+webob.multidict.MultiDict.iteritems
+webob.multidict.MultiDict.iterkeys
+webob.multidict.MultiDict.itervalues
+webob.multidict.NestedMultiDict.clear
+webob.multidict.NestedMultiDict.iteritems
+webob.multidict.NestedMultiDict.iterkeys
+webob.multidict.NestedMultiDict.itervalues
+webob.multidict.NestedMultiDict.popitem
+webob.multidict.NoVars.add
+webob.multidict.NoVars.clear
+webob.multidict.NoVars.pop
+webob.multidict.NoVars.popitem
+webob.multidict.NoVars.setdefault
+webob.multidict.NoVars.update
+webob.request.AdhocAttrMixin.__delattr__
+webob.request.AdhocAttrMixin.__getattr__
+webob.request.AdhocAttrMixin.__setattr__
+webob.request.BaseRequest.__init__
+webob.request.NoDefault
+webob.response.Response.set_cookie
+webob.response.ResponseBodyFile.write
```

Comment:
Are all of these being allowlisted because they're missing from the stub? We prefer not to allowlist anything without a comment stating why we're ignoring stubtest for this specific situation. With an allowlist this long and no comments, it's difficult to tell which objects are simply missing from the stub, and which objects stubtest is complaining about for some other reason, such as them having a different type or signature to what they have at runtime

Reply:
There is a reason for everything. The vast majority of missing stubs (all the ones ending with `_re`) are either a plain string that will be used as part of a regex that is assembled later, a marked regex string pattern or a compiled regex pattern that is used internally for parsing some part of a request/http header.
I didn't really like how inconsistently they were used, so it didn't really make sense to me to expose them to the public API, as far as I'm concerned these should all be prefixed with an underscore, but that's not up to me.

Then there is a whole slew of Python 2.7 compatibility stuff in the compat module as well as some stuff that was previously deprecated and is already removed upstream (methods such as `iteritems`, `iterkeys`, `itervalues`) as well as the `expires` argument on `set_cookie` which has been replaced by `max_age`)

I've moved some of the methods in the acceptparse from the derived class to the intermediary internal superclass, since the method signatures were the same, to reduce copy pasta, but I could move them back if that is not a good enough reason to have them on the allowlist.

I've removed `MimeAccept` since it is a deprecated API and wouldn't really interact well with the accept properties.

I've removed the `DEFAULT` parameter on the `AdhhocAttrMixin` methods purely since it seemed like internal API, I don't really see the point of ever manually setting that parameter either, which is why I removed it to make the signature less confusing.

I think that covers everything, except `webob.request.NoDefault` which I overlooked. I will fix that one.

Reply:
Okay great -- could you add all of that rationale as comments in the allowlist? I really like the way hamdanal did it when they contributed stubs for `pyserial`:

https://github.com/python/typeshed/blob/ad9c7c1d0c842ac0cb3b010ecdd9402aac5f1bd3/stubs/pyserial/%40tests/stubtest_allowlist.txt#L1-L10